vacationtime-solution/src/Code.js

```diff
@@ -0,0 +1,186 @@
+  var bigRange = sheet.getRange(beginningRow, COLUMN_NUMBER.APPROVAL,
+      numRows, 1);
+  var dropdownValues = ['APPROVED', 'NOT APPROVED', 'IN PROGRESS'];
+  var rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues)
+      .build();
+  bigRange.setDataValidation(rule);
+  bigRange.setValue('IN PROGRESS');
+}
+
+/** Creates Notified Column
+**/
+function createNotifiedColumn() {
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastRow = sheet.getLastRow();
+  var lastCol = sheet.getLastColumn();
+  var frozenRows = sheet.getFrozenRows();
+  var beginningRow = frozenRows + 1;
+  var numRows = lastRow - frozenRows;
+    
+  sheet.insertColumnAfter(lastCol);
+  sheet.getRange(1, COLUMN_NUMBER.NOTIFIED)
+      .setValue('NOTIFIED STATUS');
+  var notifiedValues = ['NOTIFIED', 'NOT NOTIFIED'];
+  var rule = SpreadsheetApp.newDataValidation().requireValueInList(notifiedValues)
+      .build();
+
+  // Sets cells to be drop-down menus.
+  var notifiedColRange = sheet.getRange(beginningRow, COLUMN_NUMBER.NOTIFIED,
+      lastRow - frozenRows, 1);
+  notifiedColRange.setDataValidation(rule);
+  notifiedColRange.setValue('NOT NOTIFIED');
+    
+  // Return column range.
+  return notifiedColRange;
+}
+
+/** 
+* Helper function that creates a calendar event for employees
+* whose vacation time has been approved.
+**/
+function createCalEvent(rangeValues, i) {
+    var rangeValues = rangeValues;
+    var i = i;
+    
+    // Get necessary values.
+    var managerEmail = rangeValues[i][COLUMN_NUMBER.MANAGER_EMAIL -
+        COLUMN_NUMBER.EMAIL];
+    var managerCal = CalendarApp.getCalendarById(managerEmail);
+    var employeeEmail = rangeValues[i][0];
+    var employeeName = rangeValues[i][1];
+    var startDate = rangeValues[i][4];
+    var endDate = rangeValues[i][5];
+    var approvalStatus = rangeValues[i][6];
+                                        
+    // Creates a calendar event.
+    var event = managerCal.createEvent('APPROVED VACATION TIME FOR ' +
+        employeeName, startDate, endDate, {
+        description: 'Your vacation time from ' +
+                                            startDate + ' to ' + endDate +
+                                            ' has been approved! Enjoy!',
+        guests: employeeEmail,
+        sendInvites: true,
+    });
+}
+
+/**
+* Checks the approval status of each employee and notifies
+* them of their status accordingly.
+**/
+function checkApproval(rangeValues, i, currentStartRow) {
+    var sheet = SpreadsheetApp.getActiveSheet();
+    var rangeValues = rangeValues;
+    var currentStartRow = currentStartRow;
+    var i = i;
+    
+    // Gets necessary values.
+    var employeeEmail = rangeValues[i][0];
+    var employeeName = rangeValues[i][1];
+    var approvalStatus = rangeValues[i][6];
+    
+    // Checks approval status.
+    if (approvalStatus == 'NOT APPROVED') {
+      EMAIL_SUBJECT = 'ERR: Vacation Time NOT Approved';
```

Comment:
This should be a constant, move it up and have:
APPROVAL_EMAIL_SUBJECT
APPROVAL_EMAIL_MESSAGE
REJECTION_EMAIL_SUBJECT
REJECTION_EMAIL_MESSAGE